test/e2e/self_provisioned_elasticsearch_test.go

```diff
@@ -0,0 +1,116 @@
+// +build self_provisioned_elasticsearch
+
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis"
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	esv1 "github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1"
+)
+
+type SelfProvisionedTestSuite struct {
+	suite.Suite
+}
+
+func(suite *SelfProvisionedTestSuite) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
+		t.Skipf("Test %s is currently supported only on OpenShift because es-operator runs only on OpenShift\n", t.Name())
+	}
+
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+
+	var err error
+	ctx, err = prepare(t)
+	if (err != nil) {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+}
+
+func (suite *SelfProvisionedTestSuite) TearDownSuite() {
+	ctx.Cleanup()
+}
+
+func TestSelfProvisionedSuite(t *testing.T) {
+	suite.Run(t, new(SelfProvisionedTestSuite))
+}
+
+func (suite *SelfProvisionedTestSuite) TestSelfProvisionedESSmokeTest() {
+	// create jaeger custom resource
+	exampleJaeger := &v1.Jaeger{
```

Comment:
To be consistent with structure, shouldn't this be moved into a separate function?